config/merge_config.go

```diff
@@ -0,0 +1,377 @@
+	agentConf.Scrubber.AddCustomSensitiveWords(strings.Split(csw, ","))
+	agentConf.Scrubber.StripAllArguments = mergeBool(dc, keyStripProcessArguments, agentConf.Scrubber.StripAllArguments)
+
+	agentConf.QueueSize = mergeInt(dc, keyQueueSize, agentConf.QueueSize)
+
+	agentConf.MaxProcFDs = mergeInt(dc, keyMaxProcFDs, agentConf.MaxProcFDs)
+
+	if mpm := mergeInt(dc, keyMaxPerMessage, agentConf.MaxPerMessage); mpm <= maxMessageBatch {
+		agentConf.MaxPerMessage = mpm
+	} else {
+		log.Warn("Overriding the configured item count per message limit because it exceeds maximum")
+	}
+
+	agentConf.DDAgentBin = mergeString(dc, keyDDAgentBin, agentConf.DDAgentBin)
+
+	agentConf.DDAgentPy = mergeString(dc, keyDDAgentPy, agentConf.DDAgentPy)
+
+	agentConf.DDAgentPyEnv = strings.Split(mergeString(dc, keyDDAgentPyEnv, ""), ",")
+
+	winAri := mergeInt(dc, keyWinArgsRefreshInterval, agentConf.Windows.ArgsRefreshInterval)
+	if winAri != 0 {
+		agentConf.Windows.ArgsRefreshInterval = winAri
+	}
+
+	agentConf.Windows.AddNewArgs = mergeBool(dc, keyWinAddNewArgs, agentConf.Windows.AddNewArgs)
+
+	if dc.IsSet(keyAdditionalEndpoints) {
+		additionalEndpoints := dc.GetStringMapStringSlice(keyAdditionalEndpoints)
+		for endpointURL, apiKeys := range additionalEndpoints {
+			u, err := url.Parse(endpointURL)
+			if err != nil {
+				return fmt.Errorf("invalid additional endpoint url '%s': %s", endpointURL, err)
+			}
+			for _, k := range apiKeys {
+				agentConf.APIEndpoints = append(agentConf.APIEndpoints, APIEndpoint{
+					APIKey:   k,
+					Endpoint: u,
+				})
+			}
+		}
+	}
+
+	// Pull additional parameters from the global config file.
+	agentConf.LogLevel = mergeString(dc, "log_level", agentConf.LogLevel)
+	agentConf.StatsdPort = mergeInt(dc, "dogstatsd_port", agentConf.StatsdPort)
+	agentConf.StatsdHost = mergeString(dc, "bind_host", agentConf.StatsdHost)
+	agentConf.Transport = ddutil.CreateHTTPTransport()
+
+	// Network related config
+	if ok, _ := isAffirmative(dc.GetString(keyNetworkTracingEnabled)); ok {
+		agentConf.EnabledChecks = append(agentConf.EnabledChecks, "connections")
+		agentConf.EnableNetworkTracing = true
+	}
+
+	agentConf.NetworkTracerSocketPath = mergeString(dc, keyNetworkUnixSocketPath, agentConf.NetworkTracerSocketPath)
+	agentConf.NetworkTracerLogFile = mergeString(dc, keyNetworkLogFile, agentConf.NetworkTracerLogFile)
+
+	agentConf.DisableTCPTracing, _ = isAffirmative(dc.GetString(keyNetworkDisableTCPTracing))
+	agentConf.DisableUDPTracing, _ = isAffirmative(dc.GetString(keyNetworkDisableUDPTracing))
+	agentConf.DisableIPv6Tracing, _ = isAffirmative(dc.GetString(keyNetworkDisableIPV6Tracing))
+
+	return nil
+}
+
+// mergeEnvironmentVariables applies overrides from environment variables to the process agent configuration
+func mergeEnvironmentVariables(dc ddconfig.Config, c *AgentConfig) {
+	var err error
+
+	// Support API_KEY and DD_API_KEY but prefer DD_API_KEY.
+	if v := os.Getenv("API_KEY"); v != "" {
```

Comment:
I think this is redundant as https://github.com/DataDog/datadog-process-agent/pull/210/files#diff-c05fe2f63536332ce41f97d262a780ddR64 will also read from the environment

Reply:
Unfortunately it will only read the `DD_API_KEY` env variable and not `API_KEY` (and we support both)